incubator/tensorflow-serving/templates/deployment.yaml

```diff
@@ -0,0 +1,64 @@
+{{- $gpuCount := .Values.gpuCount -}}
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "tensorflow-serving.fullname" . }}
+  labels:
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+    chart: {{ template "tensorflow-serving.chart" . }}
+    app: {{ template "tensorflow-serving.name" . }}
+  annotations:
+    "helm.sh/created": {{ .Release.Time.Seconds | quote }}
+spec:
+  replicas: {{ .Values.replicas }}
+  strategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      release: {{ .Release.Name | quote }}
+      app: {{ template "tensorflow-serving.name" . }} 
+  template:
+    metadata:
+      labels:
+        heritage: {{ .Release.Service | quote }}
+        release: {{ .Release.Name | quote }}
+        chart: {{ template "tensorflow-serving.chart" . }}
+        app: {{ template "tensorflow-serving.name" . }}
+    spec:
+      containers:
+        - name: serving
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: "{{ .Values.image.pullPolicy }}"
+          {{- if .Values.command }}
+          command:
+{{ toYaml .Values.command | indent 12 }}
+          {{- end }}
+          {{- if .Values.args }}
+          args:
+{{ toYaml .Values.args | indent 12 }}
+          {{- end }}
+          ports:
+            - containerPort: {{ .Values.port }}
```

Comment:
It probably doesn't make sense to make the container port configurable. Changing it would probably break things. It should suffice to make the service port configurable. I'd suggest you give the port a name and reference it by name from service and probes.